flang/lib/Lower/DirectivesCommon.h

```diff
@@ -129,8 +129,7 @@ static void processOmpAtomicTODO(mlir::Type elementType,
     // Based on assertion for supported element types in OMPIRBuilder.cpp
     // createAtomicRead
     mlir::Type unwrappedEleTy = fir::unwrapRefType(elementType);
-    bool supportedAtomicType =
-        (!fir::isa_complex(unwrappedEleTy) && fir::isa_trivial(unwrappedEleTy));
+    bool supportedAtomicType = fir::isa_trivial(unwrappedEleTy);
```

Comment:
I think we need more complex logic here. So far as I can tell this PR only fixes atomic update and capture. Using complex with atomic read (for example) still needs a TODO message.

Reply:
That's okay by me